src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,819 @@
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/Env.h>
+#include <test/jtx/envconfig.h>
+#include <xrpld/app/rdb/backend/SQLiteDatabase.h>
+#include <xrpld/rpc/CTID.h>
+#include <xrpl/protocol/ErrorCodes.h>
+#include <xrpl/protocol/STBase.h>
+#include <xrpl/protocol/STParsedJSON.h>
+#include <xrpl/protocol/jss.h>
+#include <xrpl/protocol/serialize.h>
+
+#include <optional>
+#include <tuple>
+
+namespace ripple {
+
+namespace test {
+
+class Simulate_test : public beast::unit_test::suite
+{
+    void
+    checkBasicReturnValidity(
+        Json::Value& result,
+        Json::Value& tx,
+        int expectedSequence,
+        std::string expectedFee)
+    {
+        BEAST_EXPECT(result[jss::applied] == false);
+        BEAST_EXPECT(result.isMember(jss::engine_result));
+        BEAST_EXPECT(result.isMember(jss::engine_result_code));
+        BEAST_EXPECT(result.isMember(jss::engine_result_message));
+        BEAST_EXPECT(
+            result.isMember(jss::tx_json) || result.isMember(jss::tx_blob));
+
+        Json::Value tx_json;
+        if (result.isMember(jss::tx_json))
+        {
+            tx_json = result[jss::tx_json];
+        }
+        else
+        {
+            auto unHexed = strUnHex(result[jss::tx_blob].asString());
+            SerialIter sitTrans(makeSlice(*unHexed));
+            tx_json = STObject(std::ref(sitTrans), sfGeneric)
+                          .getJson(JsonOptions::none);
+        }
+        BEAST_EXPECT(tx_json[jss::TransactionType] == tx[jss::TransactionType]);
+        BEAST_EXPECT(tx_json[jss::Account] == tx[jss::Account]);
+        BEAST_EXPECT(
+            tx_json[jss::SigningPubKey] == tx.get(jss::SigningPubKey, ""));
+        BEAST_EXPECT(
+            tx_json[jss::TxnSignature] == tx.get(jss::TxnSignature, ""));
+        BEAST_EXPECT(tx_json[jss::Fee] == tx.get(jss::Fee, expectedFee));
+        BEAST_EXPECT(
+            tx_json[jss::Sequence] == tx.get(jss::Sequence, expectedSequence));
+    }
+
+    void
+    checkBasicReturnValidity(
+        Json::Value& result,
```

Comment:
const &

Reply:
Fixed in 1c0b3a6f8ca9b038035689b68971ad5e31ba30df